src/cdk-experimental/drag-drop/drag-drop-registry.ts

```diff
@@ -0,0 +1,180 @@
+  private _globalListeners =
+      new Map<string, {handler: (event: TouchEvent | MouseEvent) => void, options?: any}>();
+
+  /**
+   * Emits the `touchmove` or `mousemove` events that are dispatched
+   * while the user is dragging a `CdkDrag` instance.
+   */
+  readonly pointerMove: Subject<TouchEvent | MouseEvent> = new Subject<TouchEvent | MouseEvent>();
+
+  /**
+   * Emits the `touchend` or `mouseup` events that are dispatched
+   * while the user is dragging a `CdkDrag` instance.
+   */
+  readonly pointerUp: Subject<TouchEvent | MouseEvent> = new Subject<TouchEvent | MouseEvent>();
+
+  constructor(
+    private _ngZone: NgZone,
+    @Inject(DOCUMENT) _document: any) {
+    this._document = _document;
+  }
+
+  /** Adds a `CdkDrop` instance to the registry. */
+  register(drop: CdkDrop);
+
+  /** Adds a `CdkDrag` instance to the registry. */
+  register(drag: CdkDrag);
+
+  register(instance: CdkDrop | CdkDrag) {
+    if (instance instanceof CdkDrop) {
+      if (!this._dropInstances.has(instance)) {
+        if (this.getDropContainer(instance.id)) {
+          throw Error(`Drop instance with id "${instance.id}" has already been registered.`);
+        }
+
+        this._dropInstances.add(instance);
+      }
+    } else {
+      this._dragInstances.add(instance);
+
+      if (this._dragInstances.size === 1) {
+        this._ngZone.runOutsideAngular(() => {
+          // The event handler has to be explicitly active, because
+          // newer browsers make it passive by default.
+          this._document.addEventListener('touchmove', this._preventScrollListener,
+              activeEventOptions);
+        });
+      }
+    }
+  }
+
+  /** Removes a `CdkDrop` instance from the registry. */
+  remove(drop: CdkDrop);
+
+  /** Removes a `CdkDrag` instance from the registry. */
+  remove(drag: CdkDrag);
+
+  remove(instance: CdkDrop | CdkDrag) {
+    if (instance instanceof CdkDrop) {
+      this._dropInstances.delete(instance);
+    } else {
+      this._dragInstances.delete(instance);
+      this.stopDragging(instance);
+
+      if (this._dragInstances.size === 0) {
+        this._document.removeEventListener('touchmove', this._preventScrollListener,
+            activeEventOptions as any);
+      }
+    }
+  }
+
+  /**
+   * Starts the dragging sequence for a drag instance.
+   * @param drag Drag instance which is being dragged.
+   * @param event Event that initiated the dragging.
+   */
+  startDragging(drag: CdkDrag, event: TouchEvent | MouseEvent) {
+    this._activeDragInstances.add(drag);
+
+    if (this._activeDragInstances.size === 1) {
+      const isTouchEvent = event.type.startsWith('touch');
+      const moveEvent = isTouchEvent ? 'touchmove' : 'mousemove';
+      const upEvent = isTouchEvent ? 'touchend' : 'mouseup';
+
+      // We explicitly bind __active__ listeners here, because newer browsers
+      // will default to passive ones for `mousemove` and `touchmove`
```

Comment:
Why do you need active handlers in the first place?

Reply:
It's because we use `preventDefault` to stop the user from scrolling the page or selecting text while they're dragging. If an event is passive, the browser will ignore the `preventDefault` calls.